sycl/doc/syclgraph/SYCLGraphUsageGuide.md

```diff
@@ -0,0 +1,515 @@
+# SYCL Graph Usage Guide and Examples
+
+This document describes recommended usage guidelines for using the
+`sycl_ext_oneapi_graph` extension (referred to as SYCL Graph in this document)
+as well as provides example code snippets for various features and usage
+scenarios.
+
+The specification for the `sycl_ext_oneapi_graph` extension can be found
+[here](../extensions/experimental/sycl_ext_oneapi_graph.asciidoc).
+
+## General Usage Guidelines
+
+The following section provides some general usage guidelines when working
+with SYCL Graph or adapting an existing SYCL application to use SYCL Graph.
+Examples here will generally only use one graph creation API ("Explicit" or
+"Record & Replay") for simplicity, but can generally be applied to either API
+unless specifically noted.
+
+### Use Host-Tasks For Host Work
+
+SYCL Graph cannot capture work done on the host in application code which does
+not go through the SYCL runtime, including direct operations (for example kernel
+submissions) through a backend API (level-zero, cuda, etc). Any work to be done
+on the host should be captured within a `sycl::host-task` as a node in the
+graph. Workloads which require a lot of host-tasks may see reduced performance
+due to host synchronization or preventing the runtime from submitted the entire
+graph to the device at once.
+
+Direct operations with backend APIs are supported in SYCL Graph through
+host-tasks using an interop handle, but it is currently not possible to obtain
+the native backend objects associated with SYCL Graph objects.
+
+Some applications may also execute host code inside of the Command Group
+Function (CGF). This is not explicitly disallowed in normal SYCL code but is
+generally discouraged. Consider the following example:
+
+```c++
+Queue.submit([&](sycl::handler& CGH){
+    // Do some host work here to prepare for the kernel to be executed
+    do_some_host_work();
+    CGH.parallel_for(...);
+});
+```
+
+In normal SYCL usage `do_some_host_work()` will be evaluated during the call to
+submit. With SYCL Graph this code will be evaluated once during the call to
+`submit()` or `command_graph::add()` but will not be evaluated on subsequent
+runs of the graph which may lead to incorrect behavior. This code should be
+placed into a `host-task` node which the kernel (or other operation) depends on,
```

Comment:
```suggestion
placed into a `host_task` node which the kernel (or other operation) depends on,
```

Reply:
Changed to just host-task as per https://github.com/intel/llvm/pull/14965#discussion_r1705376638